### PR TITLE
chore: Transform d3-* related packages to CJS

### PR DESCRIPTION
**Pull request by johannes-weber**
*Description of changes:*

We need to upgrade the `d3-color` and `d3-scale` (incl. their transitive dependencies) in the components package. With that upgrade, the format of those packages changed from CJS to ESM. To use them in our tests, we need to transform them to CJS. - This PR needs to get merged before the related one in the component package.

Related upgrade PR in the component package: https://github.com/cloudscape-design/components/pull/325

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: jest-preset.js [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+
+  // `d3-color` and `d3-scale` packages (plus their transitive dependencies) are using the ESM format.
+  // We need to transform them to have them working in tests.
+  'd3-.*',
+  'internmap',
+].join('|');
 module.exports = {
   transform: {
-    'node_modules/@cloudscape-design/.+\\.js$': require.resolve('./js-transformer'),
+    [`node_modules/(${esModules})/.+\\.js$`]: require.resolve('./js-transformer'),
```
Comment (michaeldowseza) [inline jest-preset.js:13]: Should this be an array? The [jest documentation](https://jestjs.io/docs/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object) has the definition as `object<string, pathToTransformer | [pathToTransformer, object]>`

**Review by michaeldowseza — COMMENTED** [review]
Can we double check the API usage here

File: jest-preset.js (hunk 1 above)
Comment (johannes-weber) [reply jest-preset.js:13]: It's not an array. The brackets are used to set the computed property name ([reference on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#computed_property_names)).

It's get's compiled to:
```
transform: {
   "node_modules/(@cloudscape-design|d3-.*|internmap)/.+\\.js$": require.resolve('./js-transformer')
}
```

Comment (michaeldowseza) [reply jest-preset.js:13]: All good, post holiday brain on my side 😅🚀

**Review by michaeldowseza — APPROVED** [review]
LGTM!

**Pull request merged**